Add Next.js to the Wizard (#3533)
getsentry/sentry-docs: src/wizard/javascript/nextjs.md

---
name: Next.js
doc_link: https://docs.sentry.io/platforms/javascript/guides/nextjs/
support_level: production
type: framework
---

> To use Sentry with your Next.js application, you need to use `@sentry/nextjs` (Sentry’s Next.js SDK):

Reviewer: ```suggestion
Install Sentry’s Next.js SDK using either `yarn` or `npm`:
```

```bash
yarn add @sentry/nextjs
# or
npm install --save @sentry/nextjs
```

> To [configure your app automatically](https://docs.sentry.io/platforms/javascript/guides/nextjs/#configure) (although you can also [configure it manually](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/)), you can run the sentry-wizard:
> 
> ```bash
> npx @sentry/wizard -i nextjs
> ```
> 
> It will automatically patch your application:
> 
> - create `sentry.client.config.js` and `sentry.server.config.js` with the default `Sentry.init`.
> - create `next.config.js` with the default configuration.
> - create `sentry.properties` with configuration for sentry-cli (which is used when automatically uploading source maps).

Reviewer: ```suggestion
[Configure your app automatically](https://docs.sentry.io/platforms/javascript/guides/nextjs/#configure) with the sentry wizard. It will automatically patch your application:

- Creates `sentry.client.config.js` and `sentry.server.config.js` with the default `Sentry.init`.
- Creates `next.config.js` with the default configuration.
- Creates `sentry.properties` with configuration for sentry-cli (which is used when automatically uploading source maps).

You can also [configure it manually](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/)).

```bash
npx @sentry/wizard -i nextjs
```
```

> Lastly, to capture [Next.js API route errors](https://nextjs.org/docs/api-routes/introduction), you need to wrap your handlers with a Sentry function:

Reviewer: ```suggestion
Wrap your handlers with a Sentry function to capture [Next.js API route errors](https://nextjs.org/docs/api-routes/introduction): 
```

```javascript
import { withSentry } from '@sentry/nextjs';

const handler = async (req, res) => {
  // ...
}

export default withSentry(handler);
```

> The above configuration has automatic error tracking with source maps for both JavaScript and TypeScript. To reduce the volume of performance data captured, change `tracesSampleRate` to a value between 0 and 1.

Reviewer: I'd suggest linking to the Sampling content here.  Also, `tracesSampleRate` isn't in the code sample, perhaps I'm missing it? Nonetheless, we typically direct folks to adjust this by commenting the code, not in the prose. 

From the browser wizard, for example:

  // Set tracesSampleRate to 1.0 to capture 100%
  // of transactions for performance monitoring.
  // We recommend adjusting this value in production
  tracesSampleRate: 1.0,

We recommend adjusting the value of `tracesSampleRate` in production. Learn more about configuring sampling in our [full documentation](https://docs.sentry.io/platforms/javascript/configuration/sampling/)."

> After this step, Sentry will report uncaught exceptions triggered by your Next.js application.
> 
> You can trigger your first event from your development environment by raising an exception somewhere within your application. For example, a button whose `onClick` handler throws an error:

Reviewer: ```suggestion
Then create an intentional error, so you can test that everything is working from your development environment. For example, a button whose `onClick` handler throws an error:
```

> ```jsx

Reviewer: should this be javascript?

<button type="button" onClick={() => {
    throw new Error("Sentry Frontend Error");
}}>
    Throw error
</button>
```

> Once the library is correctly initialized and an event sent, consider visiting our [complete Next.js docs](https://docs.sentry.io/platforms/javascript/guides/nextjs/).

Reviewer: We typically don't send them to docs at this point. Instead, we tell them to log in to sentry.io and check out the error. They'll be directed to docs from there.

```suggestion
If you're new to Sentry, use the email alert to access your account and complete a product tour.

If you're an existing user and have disabled alerts, you won't receive this email.
```
